### PR TITLE
Fix autocomplete logic to avoid throwing errors

### PR DESCRIPTION
**Pull request by toddjordan**
This is to fix issues surrounding autocomplete that have prompted some issues and comments
See 
#1236 
#1264

### CODE REVIEW

**Comment by toddjordan** [comment]
I merged/rebased another pr that addresses the issues, so this one just makes a few minor updates now.

File: source/localizable/tutorial/autocomplete-component.md [hunk 1]
```diff
@@ -144,11 +140,11 @@ We also define a `search` action here that is passed in to the component,
 For these actions to work, we need to modify the Mirage `config.js` file 
 to look like this, so that it can respond to our queries.
 
-```app/mirage/config.js
+```mirage/config.js
 export default function() {
   this.get('/rentals', function(db, request) {
     let rentals = [{
-        type: 'rentals',
```
Comment (locks) [inline source/localizable/tutorial/autocomplete-component.md]: type is pluralized in JSONAPI, iirc

Comment (toddjordan) [inline source/localizable/tutorial/autocomplete-component.md]: I know this works from the super rentals repo. I'll try plural and look in the spec to see

Comment (locks) [inline source/localizable/tutorial/autocomplete-component.md]: I know it's either for ED, and json:api doesn't specifically say it has to be plural, but all their [examples have it that way](http://jsonapi.org/format/#document-resource-objects). It's just a matter of consistency.

**Comment by toddjordan** [comment]
@locks good to go?

**Comment by locks** [comment]
:shipit: 

@homu r+

**Pull request merged**